### PR TITLE
[spark/stable] Added persistence for notebooks and configuration

### PR DESCRIPTION
**Pull request by jonaseck2**
*****************************************************************************************

Thank you for contributing to kubernetes/charts. Before you submit this PR we'd like to
make sure you are aware of our technical requirements and best practices:

* https://github.com/kubernetes/charts/blob/master/CONTRIBUTING.md#technical-requirements
* https://github.com/kubernetes/helm/tree/master/docs/chart_best_practices

For a quick overview across what we will look at reviewing your PR, please read
our review guidelines:

* https://github.com/kubernetes/charts/blob/master/REVIEW_GUIDELINES.md

Following our best practices right from the start will accelerate the review process and
help get your PR merged quicker.

When updates to your PR are requested, please add new commits and do not squash the 
history. This will make it easier to identify new changes. The PR will be squashed 
anyways when it is merged. Thanks.

*****************************************************************************************

### CODE REVIEW

**Comment by jonaseck2** [comment]
Based on PR #3291

/assign @scottrigby

File: stable/spark/templates/spark-zeppelin-pvc.yaml [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+      storage: {{ $persistence.Size | quote }}
+{{- if $persistence.StorageClass }}
+{{- if (eq "-" $persistence.StorageClass) }}
+  storageClassName: ""
+{{- else }}
+  storageClassName: "{{ $persistence.StorageClass }}"
+{{- end }}
+{{- end }}
+{{- end }}
+---
```
Comment (unguiculus) [inline stable/spark/templates/spark-zeppelin-pvc.yaml]: Create two separate files, please.

File: stable/spark/values.yaml [hunk 2]
```diff
@@ -64,3 +64,21 @@ Zeppelin:
...
+  Persistence:
+    Config:
+      Enabled: false
+      ## etcd data Persistent Volume Storage Class
+      ## If defined, storageClassName: <storageClass>
+      ## If set to "-", storageClassName: "", which disables dynamic provisioning
+      ## If undefined (the default) or set to null, no storageClassName spec is
+      ## set, choosing the default provisioner. (gp2 on AWS, standard on
+      ## GKE, AWS & OpenStack)
+      # storageClass: "-"
```
Comment (unguiculus) [inline stable/spark/values.yaml]: `# StorageClass: "-"`

File: stable/spark/values.yaml [hunk 3]
```diff
@@ -64,3 +64,21 @@ Zeppelin:
...
+      ## Set default PVC size
+      Size: 10G
+      ## Set default PVC access mode: https://kubernetes.io/docs/concepts/storage/persistent-volumes/#access-modes
+      AccessMode: ReadWriteOnce
+    Notebook:
+      Enabled: false
```
Comment (unguiculus) [inline stable/spark/values.yaml:85]: StorageClass missing

File: stable/spark/README.md [hunk 4]
```diff
@@ -76,22 +76,28 @@ The following tables lists the configurable parameters of the Spark chart and th
...
+| `Zeppelin.Ingress.Annotations`           | annotations for the ingress          | `{}`                                       |
+| `Zeppelin.Ingress.Path`                  | the ingress path                     | `/`                                        |
+| `Zeppelin.Ingress.Hosts`                 | a list of ingress hosts              | `[zeppelin.example.com]`                   |
+| `Zeppelin.Ingress.Tls`                   | a list of [IngressTLS](https://v1-8.docs.kubernetes.io/docs/api-reference/v1.8/#ingresstls-v1beta1-extensions) items | `[]`
+| `Zeppelin.ServiceType `                  | Kubernetes Service type              | `LoadBalancer`                             |
+| `Zeppelin.Persistence.Config.Enabled`    | Enable Persistence for configuration | `false`                                    |
+| `Zeppelin.Persistence.Config.Size`       | Configuration Persistence Size       | `10G`                                      |
+| `Zeppelin.Persistence.Config.Size`       | Configuration Persistence AccessMode | `ReadWriteOnce`                            |
```
Comment (unguiculus) [inline stable/spark/README.md:97]: StorageClass missing.

**Review by unguiculus — CHANGES_REQUESTED** [review]

**Comment by jonaseck2** [comment]
What are we waiting for?

**Review by unguiculus — APPROVED** [review]

**Comment by VinceMD** [comment]
It's broken:

https://github.com/helm/charts/issues/7478
https://github.com/helm/charts/issues/6784 

Even though it's in stable

**Pull request merged**